### DISCUSSION TITLE
How to perform case insensitive / alphabetical order on results?

### DISCUSSION THREAD

**Discussion by xzilja** [supabase/supabase#4610] [Questions] [answered]
I have a table of products that has a `name` column. I expected

```js
.order('name', { ascending: true })
```

To return alphabetically ordered list of products, however it doesn't seem to be the case. General order of items is correct, but some times I get few products starting with letter `c` in between `a` and `A` i.e. it feels like this order is case sensitive?

Also some products with symbols in their names like `.` or `?` (can be in the middle, start or end of the name) squeeze through in incorrect order.

Finally mixed names with upper and lowercase letters are misplaced as well.

As far as I can see there is nothing on `.order()` that can allow specifying for it to be case / symbol insensitive? So I decided to ask for help here to see whether this is a bug or I am missing something to perform correct alphabetical order.

**Comment by steve-chavez** [comment, marked as answer]
You could do this with:

1) A `citext`(case insensitive text) type. First enable the `citext` extension through the UI and then:
 
  ```sql
  ALTER TABLE table_name ALTER COLUMN name TYPE citext;
  ```

2) A computed column, see this previous discussion: https://github.com/supabase/supabase/discussions/2402
